client/lib/login/index.js

```diff
@@ -120,26 +117,45 @@ export function getSignupUrl( currentQuery, currentRoute, oauth2Client, locale,
 		if ( wccomFrom ) {
 			oauth2Params.set( 'wccom-from', wccomFrom );
 		}
-		signupUrl = `${ signupUrl }/wpcc?${ oauth2Params.toString() }`;
+		return `${ signupUrl }/wpcc?${ oauth2Params.toString() }`;
 	}
 
 	if ( oauth2Client && isJetpackCloudOAuth2Client( oauth2Client ) ) {
 		const oauth2Params = new URLSearchParams( {
 			oauth2_client_id: oauth2Client.id,
 			oauth2_redirect: redirectTo,
 		} );
-		signupUrl = `${ signupUrl }/wpcc?${ oauth2Params.toString() }`;
+		return `${ signupUrl }/wpcc?${ oauth2Params.toString() }`;
+	}
+
+	if ( oauth2Client ) {
+		const oauth2Params = new URLSearchParams( {
+			oauth2_client_id: oauth2Client.id,
+			oauth2_redirect: redirectTo,
+		} );
+		return `${ signupUrl }/wpcc?${ oauth2Params.toString() }`;
+	}
+
+	if ( signupFlow ) {
+		if ( redirectTo ) {
+			const params = new URLSearchParams( {
+				redirect_to: redirectTo,
+			} );
+			return `${ signupUrl }/${ signupFlow }?${ params.toString() }`;
+		}
+		return `${ signupUrl }/${ signupFlow }`;
```

Comment:
Moved signupFlow down to here, maintaining redirectTo pass through for consistency, I can't actually see anything that uses it other than 3-4 instances in trunk that are only using it to bypass /start (they're setting signup_flow to account i.e. what we would return with the redirect_to anyway.)